engineapi/backups.go

```diff
@@ -277,32 +277,52 @@ func (e *Engine) SnapshotBackup(backupName, snapName, backupTarget, backingImage
 	// get environment variables if backup for s3
 	envs, err := getBackupCredentialEnv(backupTarget, credential)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 	output, err := e.ExecuteEngineBinaryWithoutTimeout(envs, args...)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 	backupCreateInfo := BackupCreateInfo{}
 	if err := json.Unmarshal([]byte(output), &backupCreateInfo); err != nil {
-		return "", err
+		return "", "", err
 	}
 
 	logrus.Debugf("Backup %v created for volume %v snapshot %v", backupCreateInfo.BackupID, e.Name(), snapName)
-	return backupCreateInfo.BackupID, nil
+	return backupCreateInfo.BackupID, backupCreateInfo.ReplicaAddress, nil
 }
 
-func (e *Engine) SnapshotBackupStatus() (map[string]*longhorn.BackupStatus, error) {
-	args := []string{"backup", "status"}
+func (e *Engine) SnapshotBackupStatus(backupName, replicaAddress string) (*longhorn.EngineBackupStatus, error) {
+	args := []string{"backup", "status", backupName}
+	if replicaAddress != "" {
+		args = append(args, "--replica", replicaAddress)
+	}
+
 	output, err := e.ExecuteEngineBinary(args...)
 	if err != nil {
 		return nil, err
 	}
-	backups := make(map[string]*longhorn.BackupStatus)
-	if err := json.Unmarshal([]byte(output), &backups); err != nil {
+
+	engineBackupStatus := &longhorn.EngineBackupStatus{}
```

Comment:
Because `replicaAddress` is possible to be empty string, shouldn't we still have getting all backup status in this existing case?

Reply:
If `replicaAddress `is empty, the longhorn engine will find the corresponding replica by looping all the replicas. You could refer to the longhorn engine [PR](https://github.com/longhorn/longhorn-engine/blob/77d1117f3c917be41951a1c6d4b6a51d1a2ff550/app/cmd/backup.go#L117-LL138).